key.lua

```diff
@@ -0,0 +1,29 @@
+
+minetest.register_tool("telemosaic:key", {
+	description = "Telemosaic Key",
+	inventory_image = "telemosaic_key.png",
+	stack_max = 1,
+	groups = {not_in_creative_inventory = 1},
+	on_use = function(itemstack, player, pointed_thing)
+		local pos = pointed_thing.under
+		if pointed_thing.type == "node" then
+			-- Use key to toggle telemosaic on/off
+			local state = telemosaic.get_state(pos)
+			if state == "active" then
+				telemosaic.set_state(pos, "disabled")
+			elseif state == "disabled" then
+				telemosaic.set_state(pos, "active")
+			end
+		elseif player:get_player_control().sneak then
+			-- Get rid of key
+			return ItemStack("default:mese_crystal_fragment")
+		end
```

Comment:
I think after last commit it would be even more important to not do this as now sneak is used to preserve key when using it on telemosaic, fail reaching beacon and instead of preserving key which was what you meant to do you will instead destroy key.

Reply:
Except one is done with left-click, and the other is done with right-click ;)

Reply:
Oh well, yes. Then only earlier comments about this applies (accidental deletion)

Reply:
Now that I think of it, the number of times one would need to destroy a key "manually" is very few, so probably not worth the risk of accidental deletion, and there's still the craft recipe anyway.